src/GogeDao.sol

```diff
@@ -1,1087 +1,793 @@
+    /// @notice The maximum time needed for a new poll -> default is 60 days.
+    uint256 public maxPeriod = 60 days;
+    /// @notice The minimum balance of governance token the author of a poll must be holding at the time of creation.
+    uint256 public minAuthorBal = 10_000_000 ether;
+    /// @notice The maximum amount of polls an author can have active at any given time.
+    uint256 public maxPollsPerAuthor = 1;
+    /// @notice The threshold that must be met to pass a poll
+    uint256 public quorum = 50;
 
-import "./libraries/Libraries.sol";
-import "./extensions/Ownable.sol";
-import "./extensions/IGogeERC20.sol";
+    /// @notice Amount of BNB held for marketing purposes.
+    uint256 public marketingBalance;
+    /// @notice Amount of BNB held for team pay
+    uint256 public teamBalance;
 
+    /// @notice Bool if gate keeping is enabled.
+    bool public gatekeeping = true;
+    /// @notice Bool if createPoll is callable.
+    bool public createPollEnabled;
 
-contract GogeDAO is Ownable {
-    using SafeMath for uint256;
+    /// @notice Array of team member addresses.
+    address [] public teamMembers;
+    /// @notice Array of active polls -> array of poll nums.
+    uint256 [] public activePolls;
     
-    address public governanceTokenAddr;
-    uint256 public pollNum;
-    uint256 public minPeriod = 86400;
+    /// @notice Double Mapping of pollNum to amount of votes per voter.
     mapping(uint256 => mapping(address => uint256)) public polls;
+    /// @notice Mapping of pollNum to array of addresses of voters
     mapping(uint256 => address[]) public voterLibrary;
+    /// @notice Mapping of pollNum to amount of total votes per poll.
     mapping(uint256 => uint256) public totalVotes;
-    mapping(uint256 => uint256) public historicalTally;
-    mapping(uint256 => uint256) public pollStartTime;
-    mapping(uint256 => uint256) public pollEndTime;
-    mapping(uint256 => bool) public passed;
-    bool public veto = true;
-    mapping(address => bool) public vetoAuthority;
-    address [] public teamMembers;
-    string [] public actions;
-    uint256 public marketingBalance;
-    uint256 public teamBalance;
+    /// @notice Mapping of pollNum to poll author's address.
+    mapping(uint256 => address) public pollAuthor;
 
-    uint256 public quorum = 50;
+    /// @notice Mapping of pollNum to whether or not a poll has been passed (bool).
+    mapping(uint256 => bool) public passed;
+    /// @notice Mapping of address to whether or not it is a gate keeper.
+    mapping(address => bool) public gatekeeper;
+    
+    /// @notice Mapping of address to array of pollNums it has outstanding votes for.
+    mapping(address => uint256[]) public advocateFor;
 
-    //PollTypes.PollType public PollType;
+    /// @notice Mapping of pollNum to it's specified pollType.
+    mapping(uint256 => PollType) public pollTypes;
+    /// @notice Mapping of pollNum to it's specified metadata.
+    mapping(uint256 => Proposal) public pollMap;
```

Comment:
updated pollMap name to proposals. Personally, I like having a corresponding mapping that is updated specifically for each poll type instead of having it in the same constructor. I think this way is cleaner